ethereum/contracts/state-transition/chain-deps/facets/Mailbox.sol

```diff
@@ -245,72 +353,69 @@ contract MailboxFacet is Base, IMailbox {
 
         // The L1 -> L2 transaction may be failed and funds will be sent to the `_refundRecipient`,
         // so we use `msg.value` instead of `_l2Value` as the bridged amount.
-        _verifyDepositLimit(msg.sender, msg.value);
-        canonicalTxHash = _requestL2Transaction(
-            sender,
-            _contractL2,
-            _l2Value,
-            _calldata,
-            _l2GasLimit,
-            _l2GasPerPubdataByteLimit,
-            _factoryDeps,
-            false,
-            _refundRecipient
-        );
+        _verifyDepositLimit(_sender, _msgValue);
+
+        // Here we manually assign fields for the struct to prevent "stack too deep" error
+        WritePriorityOpParams memory params;
+
+        params.sender = sender;
+        params.l2Value = _l2Value;
+        params.contractAddressL2 = _contractL2;
+        params.l2GasLimit = _l2GasLimit;
+        params.l2GasPricePerPubdata = _l2GasPerPubdataByteLimit;
+        params.refundRecipient = _refundRecipient;
+
+        canonicalTxHash = _requestL2Transaction(_msgValue, params, _calldata, _factoryDeps, false);
     }
 
     function _verifyDepositLimit(address _depositor, uint256 _amount) internal {
-        IAllowList.Deposit memory limitData = IAllowList(s.allowList).getTokenDepositLimitData(address(0)); // address(0) denotes the ETH
+        IAllowList.Deposit memory limitData = IAllowList(chainStorage.allowList).getTokenDepositLimitData(address(0)); // address(0) denotes the ETH
         if (!limitData.depositLimitation) return; // no deposit limitation is placed for ETH
 
-        require(s.totalDepositedAmountPerUser[_depositor] + _amount <= limitData.depositCap, "d2");
-        s.totalDepositedAmountPerUser[_depositor] += _amount;
+        require(chainStorage.totalDepositedAmountPerUser[_depositor] + _amount <= limitData.depositCap, "d2");
+        chainStorage.totalDepositedAmountPerUser[_depositor] += _amount;
+    }
+
+    // for setting chainID in VM
+    function requestL2TransactionProof(
+        WritePriorityOpParams memory _params,
+        bytes calldata _calldata,
+        bytes[] calldata _factoryDeps,
+        bool _isFree
+    ) external onlyStateTransition returns (bytes32 canonicalTxHash) {
+        canonicalTxHash = _requestL2Transaction(0, _params, _calldata, _factoryDeps, _isFree);
     }
 
     function _requestL2Transaction(
```

Comment:
Should we strive to have different transaction hashes for different chains? Except for pre-EIP155 transactions, the chainId is included with Ethereum transactions

Reply:
Probably makes sense to not add those to not introduce additional breaking changes

Reply:
I don't think it is necessary here. Transactions cannot be replayed here, because it is the priority queue that checks the transactions here. But for hyperbridges we will have to have a nullifier tree, and there it will matter to have different hashes. Otherwise it could happen that a transaction has the same hash as another.... But we can just include the txID, and the sender ChainID, then we will not have to introduce any breaking changes. So I think it is good for now.